hibernation-setup-tool.c

```diff
@@ -1434,13 +1434,13 @@ static bool is_cold_boot(void)
     if (lock_file_path) {
         unlink(hibernate_lock_file_name);
 
-        if (access(lock_file_path, F_OK) < 0)
-            return true;
+        if (!access(lock_file_path, F_OK))
```

Comment:
if access system call returns true, is it appropriate to return that it cold booted? Any scenario that this call would return true?

Reply:
This tripped me up but access returns 0 upon successful access (other examples of this in the code). This check determines if we can still access the temporary file: lock_file_path (/tmp/hibernation-setup-tool/hibernatedXXXXXX). It finds the file path by reading from the symbolic link hibernate_lock_file_name (/etc/hibernation-setup-tool.last_hibernation). If it can read from it, it knows we haven't cold booted and so returns false. If it can't read from it, it says we've cold booted and returns true.